arangod/Utils/SupportInfoBuilder.cpp

```diff
@@ -0,0 +1,762 @@
+      if (manager != nullptr) {
+        agents = manager->endpoints().size();
+      }
+
+      result.add("agents", VPackValue(agents));
+      result.add("coordinators", VPackValue(coordinators));
+      std::string dbServersKey = "dbServers";
+      std::string shardsStatitsics = "shards";
+      if (isTelemetricsReq) {
+        dbServersKey = "db_servers";
+        shardsStatitsics = "shards_statistics";
+      }
+      result.add(dbServersKey, VPackValue(dbServers));
+
+      if (ServerState::instance()->isCoordinator()) {
+        result.add(VPackValue(shardsStatitsics));
+        ci.getShardStatisticsGlobal("", result);
+      }
+
+      if (isTelemetricsReq) {
+        result.add("date", VPackValue(timeString));
+        result.add("databases", VPackValue(VPackValueType::Array));
+        if (!dbInfoBuilder.isEmpty()) {
+          addDatabaseInfo(result, dbInfoBuilder.slice(), server);
+        }
+        result.close();
+      }
+
+      result.close();  // deployment
+      if (!isTelemetricsReq) {
+        result.add("date", VPackValue(timeString));
+      }
+
+    } else {
+      // DB server or other coordinator
+      result.add("host", hostInfo.slice());
+      if (isTelemetricsReq && !ServerState::instance()->isCoordinator()) {
+        VPackBuilder serverInfo;
+        buildDbServerDataStoredInfo(serverInfo, server);
+        result.add("databases", serverInfo.slice().get("databases"));
+      }
+    }
+  }
+  result.close();
+}
+
+void SupportInfoBuilder::buildHostInfo(VPackBuilder& result,
+                                       ArangodServer& server,
+                                       bool isTelemetricsReq) {
+  bool const isActiveFailover =
+      server.getFeature<ReplicationFeature>().isActiveFailoverEnabled();
+
+  result.openObject();
+
+  if (isTelemetricsReq || ServerState::instance()->isRunningInCluster() ||
+      isActiveFailover) {
+    auto serverId = ServerState::instance()->getId();
+    if (isTelemetricsReq) {
+      bool isSingleServer = ServerState::instance()->isSingleServer();
+      if (isSingleServer) {
+        auto const serverIdSingle = ServerIdFeature::getId().id();
+        result.add("id",
+                   VPackValue("single_" + std::to_string(serverIdSingle)));
+      } else {
+        normalizeKeyForTelemetrics(serverId);
+        result.add("id", VPackValue(serverId));
+      }
+    } else {
+      result.add("id", VPackValue(serverId));
+    }
+    result.add("alias", VPackValue(ServerState::instance()->getShortName()));
+    result.add("endpoint", VPackValue(ServerState::instance()->getEndpoint()));
+  }
+
+  result.add("role", VPackValue(ServerState::roleToString(
+                         ServerState::instance()->getRole())));
+  result.add("maintenance",
+             VPackValue(ServerState::instance()->isStartupOrMaintenance()));
+  containers::FlatHashMap<std::string_view, std::string_view> keys;
+
+  if (isTelemetricsReq) {
+    keys["readOnly"] = "read_only";
```

Comment:
ok, why?
Instead of doing all this if...then...else stuff here in the server code, can't we just make the gathering endpoint use the attribute names that we had already in use. That would simplify a lot.

Reply:
I agree, but, but was told no

Reply:
Though I've been getting OK from the endpoint, so it seems it can parse without these rules